### PR TITLE
Fix for #254 -- accept dict dtype in `is_signal`

### LINKED ISSUE #254: `is_signal` Type Checking Failure
### `is_signal` Type Checking Failure 

**Description:**
The `is_signal` method fails due to incorrect type checking. Specifically, `d["metadata"]` is expected to be a list but is instead a dictionary, causing the check to fail when `is_signal` is invoked.

**Steps to Reproduce:**
The following code snippet has this the issue, revealing that `d["metadata"]` is a dictionary, not a list, as expected by `is_signal` which is modified for debugging below.

```python
d = Signal(
    data=create_signal_data(i + 1j * q),
    metadata=create_modulated_rf_metadata(
        sample_rate=1024,
        num_samples=4096,
        complex=True,
        snr=100,
        samples_per_symbol=32  # samp_rate/symbol_rate
    )
)

if not isinstance(d, dict):
    print("Error: `d` is not a dictionary.")

if "data" not in d.keys() or "metadata" not in d.keys():
    print("Error: `data` or `metadata` key missing in `d`.")

print(d["metadata"], type(d["metadata"]))
if not isinstance(d["metadata"], list):
    print("Error: `metadata` is not a list as expected.")

print(is_signal_data(d["data"]) and all(
    [is_signal_metadata(m) for m in d["metadata"]]
))
```

**Expected Behavior:**
The `is_signal` method should pass this check without failure, except that `create_modulated_rf_metadata` returns the incorrect type. 

**Proposed Solution:**
Wrap the dictionary in a list within `create_modulated_rf_metadata` to meet the expected structure. This will align the data type with the `is_signal` method’s requirements, resolving the type mismatch. Or, just expect a dictionary.

### PR DESCRIPTION
**Pull request by IsaiahHarvi**
Closes #254

### CODE REVIEW

**Comment by IsaiahHarvi** [comment]
Alternatively, could update `is_signal` line 235 to:

   `if not isinstance(d["metadata"], dict):`

**Comment by ereoh** [comment]
Thanks for the issue and PR. I believe the reason for checking if the metadata is a list comes from our transforms? I am cautious about breaking some of our other code by changing the `create_modulated_rf_metadata` method. Would the below fix work for you?

`utils/types.py`, line 235
```
if not isinstance(d["metadata"], dict) or isinstance(d["metadata"], list):
```

This fix is pretty temporary FYI. Just a heads up, we are working on a huge rehaul/rewrite of the code in a few releases, which would hopefully fix a lot of the confusing code and bugs. In addition better open-source collaboration resources like an updated Wiki.

**Comment by IsaiahHarvi** [comment]
Yeah, that's definitely an acceptable workaround; I went ahead and adjusted accordingly.  

Happy to hear about the rewrite! I'm excited to see what's in the pipeline.

**Comment by ereoh** [comment]
Great! Also small correction (also my bad too):
``
if not isinstance(d["metadata"], dict) and not isinstance(d["metadata"], list):
```

File: torchsig/utils/types.py [hunk 1]
```diff
@@ -231,8 +231,8 @@ def is_signal(d: Signal) -> bool:
 
     if "data" not in d.keys() or "metadata" not in d.keys():
         return False
-
-    if not isinstance(d["metadata"], list):
+
```
Comment (IsaiahHarvi) [inline torchsig/utils/types.py]:
```suggestion

```

**Comment by IsaiahHarvi** [comment]
Oh no problem, good catch!

**Pull request merged**